### PR TITLE
handle update twin w seed

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

Edit update twin

### Changes

edit update twin to handle hex string case

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/219

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings

### CODE REVIEW

File: packages/grid_client/src/clients/tf-grid/twins.ts [hunk 1]
```diff
@@ -10,10 +10,22 @@ class Twins {
...
+
+    if (bip39.validateMnemonic(secret)) {
+      const seed = bip39.mnemonicToSeedSync(secret);
+      privKey = new Uint8Array(seed).slice(0, 32);
+      pk = "0x" + Buffer.from(secp.getPublicKey(privKey, true)).toString("hex");
+    } else {
+      if (secret.startsWith("0x")) {
+        secret = secret.substring(2);
+      }
+      privKey = Buffer.from(secret, "hex");
```
Comment (xmonader) [inline packages/grid_client/src/clients/tf-grid/twins.ts:25]: this is not protected

File: packages/grid_client/src/clients/tf-grid/twins.ts [hunk 2]
```diff
@@ -10,10 +10,22 @@ class Twins {
...
-    const privKey = new Uint8Array(seed).slice(0, 32);
-    const pk = "0x" + Buffer.from(secp.getPublicKey(privKey, true)).toString("hex");
+  getPublicKey(secret: string) {
+    let privKey;
+    let pk;
+
+    if (bip39.validateMnemonic(secret)) {
+      const seed = bip39.mnemonicToSeedSync(secret);
+      privKey = new Uint8Array(seed).slice(0, 32);
+      pk = "0x" + Buffer.from(secp.getPublicKey(privKey, true)).toString("hex");
```
Comment (AhmedHanafy725) [inline packages/grid_client/src/clients/tf-grid/twins.ts:20]: this line is repeated in both cases

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
this should be also handled in the rmb direct client

File: packages/grid_client/src/clients/tf-grid/twins.ts (hunk 1 above)
Comment (zaelgohary) [reply packages/grid_client/src/clients/tf-grid/twins.ts:25]: This is already handles by the third party package that encrypts the private key. 

image

Comment (zaelgohary) [reply packages/grid_client/src/clients/tf-grid/twins.ts]: I added seed validation before generating the public key.

**Review by Mahmoud-Emad — APPROVED** [review]

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
why do we have too much changes in yarn.lock?

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**